### DISCUSSION TITLE
More shadowCss examples and help please

### DISCUSSION THREAD

**Discussion by dev4qa** [selenide/selenide#1725] [Q&A] [answered]
Hi

It would be good with some more examples on `shadowCss` navigation, and some help :-) .
I am familiar with normal navigation but not shadow css navigation.
I hade a look at the /src/test/java/integration/ShadowElementTest.java to mimic an expression on my own page.
However, I cannot find a working syntax for matching or selecting a `/product/product_xyz` (or even better maybe , the text) inside a shadow section using selenide, in my own page.

Any ideas how the syntax should be in Selenide?

In chrome browser F12 I see this structure similar to this:

 
```html
<div>
  <company-grid>
    <company-heading>
      <b2b-product heading="productname">
        #shadow root (open) // Comment: first, topmost, shadow root
      <xyz-frame class="xyzclass">
        <a href="/product/product_xyz">
          <company-heading class="heading-class>
            #shadow root (open) // Comment: second, deeper shadow root
            <h3>
              "Product A100"
```

 For example  I have unsuccessfully  tried with :

`$("body").find(shadowCss("//a[@href='/product/product_xyz']","#shadow-root"))`

Any hints or ideas much appreciated and welcome , thank you.

**Comment by asolntsev** [comment, marked as answer]
@dev4qa First of all, method `shadowCss` accepts CSS selectors as parameters. 
But you are passing `//a[@href='/product/product_xyz']` which is NOT a valid CSS selector. It rather looks like xpath. 

Now, the second parameter should be CSS selector of an existing web element (which is the root of shadow dom). 
You are passing `#shadow-root`, but there is no web element with `id=shadow-root` in your html. 


RTFT!
https://github.com/selenide/selenide/blob/master/src/test/java/integration/ShadowElementTest.java
